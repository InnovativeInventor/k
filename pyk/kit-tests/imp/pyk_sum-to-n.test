kit-load pyk_sum-to-n_summary

kit-clean
kit-set verbose

mkdir -p "${summary_dir}"
kompile --backend haskell -o "${summary_dir}/kompiled" --emit-json ../../k-files/imp-verification.k
kit-init pyk_sum-to-n pyk_sum-to-n-spec.k

list-cfgs

case-split '#init'       'N ==Int 1'   --alias-true init-1   --alias-false init-not-1
case-split '@init-not-1' 'N ==Int 10'  --alias-true init-10

show-edge @init-not-1,@init-10
show-node @init-10

kit-set profile

build-edges "@init-1,#target"
verify-edges --edges "@init-1,#target"

build-edges "@init-10,#target"
verify-edges --edges "@init-10,#target"

show-cfg
