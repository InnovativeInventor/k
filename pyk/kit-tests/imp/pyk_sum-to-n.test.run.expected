+ init pyk_sum-to-n pyk_sum-to-n-spec.k
+ list-cfgs
PYK_SUM-TO-N-SPEC.main
+ case-split #init N ==Int 1 --alias-true init-1 --alias-false init-not-1
+ case-split @init-not-1 N ==Int 10 --alias-true init-10
+ show-edge @init-not-1,@init-10
e3a5fd..bf2324 (expanded, @init-not-1)
│ N ==Int 10
a1a89d..d4e478 (frontier, leaf, @init-10)

+ show-node @init-10
<T>
  <k>
    int $n , $s , .Ids ; $n = N ; $s = 0 ; while ( ! $n <= 0 ) { $s = $s + $n ; $n = $n + -1 ; }
  </k>
  <state>
    .Map
  </state>
</T>
#And { true #Equals N ==Int 10 }
#And { true #Equals notBool ( N ==Int 1 ) }
#And { true #Equals N >=Int 0 }
+ build-edges @init-1,#target
+ verify-edges --edges @init-1,#target
+ build-edges @init-10,#target
+ verify-edges --edges @init-10,#target
+ show-cfg
5fa35d..e09951 (init, expanded)
┣━ fdeda8..09e172 (expanded, @init-1)    N ==Int 1
┃   │  [1m[32m(verified)[0m[0m
┃   │  (-1 steps)
┃   └  6a6e72..499ac5 (target, leaf)
┃
┗━ e3a5fd..bf2324 (expanded, @init-not-1)    notBool ( N ==Int 1 )
    ┣━ a1a89d..d4e478 (expanded, @init-10)    N ==Int 10
    ┃   │  [1m[32m(verified)[0m[0m
    ┃   │  (-1 steps)
    ┃   └  6a6e72..499ac5 (target, leaf)
    ┃
    ┗━ [1me4765a..0dc322 (frontier, leaf)[0m    notBool ( N ==Int 10 )


